libs/utils/utils.js

```diff
@@ -144,6 +144,23 @@ export const [setConfig, getConfig] = (() => {
         // eslint-disable-next-line no-console
         console.log('Invalid or missing locale:', e);
       }
+      if (Array.isArray(config.contentRoot)) {
+        const pathSplit = pathname.split('/'); 
+        const contentRoots = config.contentRoot;   
+        if (contentRoots.includes(pathSplit[1])) {
+          config.contentRoot = `/${pathSplit[1]}`;
+        } else if (contentRoots.includes(pathSplit[2])) {
+          if (config.locale.prefix.length > 0) {
+            config.contentRoot = `/${pathSplit[2]}`;
+          } else {
+            config.contentRoot = `/${pathSplit[1]}/${pathSplit[2]}`;
+          }
+        } else if (!config.locale.prefix.length > 0 && !contentRoots.includes(pathSplit[1]) && pathSplit[2]) {
+          config.contentRoot = `/${pathSplit[1]}`;
+        } else {
+          config.contentRoot = '';
+        }
+      }
```

Comment:
This is complicated to read due to many if-else statements. If this is only way to achieve the logic, it will need some comments.

Reply:
It seems the new logic isn't covered by test.

Reply:
It looks like coverage is there generically. Do you think we should write extensive tests for this?